### ISSUE TITLE
Admin users should not be able to see private repositories on users profile page

### ISSUE THREAD

**Issue by lautriva** [go-gitea/gitea#27258]
### Description

I understand that admin users have all access

But, when logged as an administrator
Private repos of other users should not be shown on their profile page `https://git.example.com/{USERNAME}` and only shown at `https://git.example.com/admin/repos`

## Example
User A is an admin. User B is a regular user

User B have many repos: some private, some shared (with A as a collaborator) and some public ones

#### Expected behavior
User A should only sees B's public or collaborated repositories on B's public page

#### Actual Behavior
All B's repositories are shown

----

### Gitea Version

1.20.4

### Can you reproduce the bug on the Gitea demo site?

No

### How are you running Gitea?

Docker installation with docker-compose

### Database

MySQL/MariaDB

**Comment by wfjake** [comment]
Seems to be duplicate of #3442. With most software deployments like Gitea, OpenProject, JIRA, Confluence, typically people with administrator rights will have two accounts - their main, non-admin account and an admin account which is used exclusively for administration purposes.

Using an admin account as daily driver is kind of like logging into servers as 'root'. If computer is infected or they accidentally run command, there is lot of potential for damage.

**Comment by dav64** [comment]
I think the code to check that is already there
For example in the main page, the `Repositories` list (see below) seems to honor the permissions (I don't see people's private repos)
I think its just a bug because the same permissions aren't applied between those 2 pages

![image](https://github.com/go-gitea/gitea/assets/10651285/816fc741-323e-42ca-afbe-d74db36a74c1)

**Comment by lunny** [comment]
Administrators should find the repositories from admin panel features.

**Comment by dav64** [comment]
Yes I know this one, in my example I referred it as `https://git.example.com/admin/repos`
I know and I'm ok with the fact that here all instance repos are shown :slightly_smiling_face: 

My problem is when I check on user profile page `https://git.example.com/{USERNAME}`, their private repos are shown here but they shouldn't be

**Comment by LBlaboon** [comment]
Just throwing my two cents in here - I would also like to see private repos hidden from admin users unless they are specifically viewing one of the `/admin/*` pages. This would put Gitea in line with other self-hosted git solutions.

For example, at my work I am a site admin of our Github Enterprise deployment, but even as such I cannot see any repos that I don't have direct access to. I can only see them through the admin interface which requires re-authentication on each visit. I understand the re-auth portion of this may be tricky or not feasible with how Gitea works, but just hiding private repos from admin users on non-admin endpoints would be enough for me, IMO.

**Comment by baptaegir** [comment]
Don't know it is directly linked, but the gitlab way might help : 
You can have admin account acts like standard user account by default. When accessing admin settings/view, it requires a new authentication.
